.github/workflows/code-quality.yml

```diff
@@ -100,6 +100,12 @@ jobs:
           clang-format --version
           cpplint --version
 
+      - name: CMake lint
+        shell: bash
+        run: |
+          python -m pip install cmakelang pyyaml
+          find . \( -name "CMakeLists.txt" -o -wholename "*/cmake/*.cmake" \) | xargs cmake-lint
```

Comment:
[question] Is there a reason to limit the `*.cmake` search to only files under the `cmake` directory, whereas `CMakeLists.txt` are searched throughout the repo?

Perhaps `-name "*.cmake"` would be better (we can add exclusions if there's anywhere we don't want it to search).

Reply:
There was a lot of stuff in the conan directory it was picking up initially, I'll try an exclusion first approach instead.

Reply:
The exclusion of the hidden folders solved this, https://github.com/OpenAssetIO/OpenAssetIO/pull/811/commits/c37e253dcc71eb7d0c1f9c5c67b6a2002b21aa5e